### PR TITLE
chall11 | estina

### PR DESCRIPTION
**Pull request by xplore93**
My solution for this problem.
First, I make a strsep by "\n".
Then I check if first char isdigit() and if so, make asprintf() into my result string. If the first char is not a digit, I assume that it's "goto" and pass through all next strings searching the label. If so, make asprintf() into my result string. If not, throw and "Infinite Loop!".

### CODE REVIEW

File: chall11/estina.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	char	*duplicate = strdup(code);
+	char	*token = strsep(&duplicate, "\n");
+
+	while (token)
+	{
+		if (isdigit(token[0]))
+			put_line(&response, token);
+		else
+		{
+			label = atoi(token + 4);
```
Comment (aguimbao) [inline chall11/estina.c:52]: In your last loop, you call `atoi()` here with an empty string, this causes a buffer-overflow (you can, and you should always) verify this with `gcc`'s `-g` and `-fsanitize=address` options. It's a shame cause you submitted a really nice solution.

File: chall11/estina.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			}
+		}
+		line = strsep(&token, " ");
+	}
+}
+
+static char	*return_infinite(char **duplicate, char **response)
+{
+	if (*response)
+		free(*response);
```
Comment (aguimbao) [inline chall11/estina.c:34]: Don't worry about leaks with the challenges, but good practice :)

File: chall11/estina.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			label = atoi(token + 4);
+			if (loop == -1)
+				loop = label;
+			else if (loop == label)
+				return (return_infinite(&duplicate, &response));
+			token = strsep(&duplicate, "\n");
+			while (token)
+			{
+				if (isdigit(token[0]))
+					if (atoi(token) == label)
```
Comment (aguimbao) [inline chall11/estina.c:61]: Maybe `atoi()` was a bit unnecessary ?

**Review by aguimbao — COMMENTED** [review]
I don't have much to say about your solution, it's very optimal and you have really nice practices as usual !

It's a shame you didn't sanitise addresses with `fsanitize` :( Cause you pass all the tests otherwise. But your extra `atoi()` makes you overflow on every test.

File: chall11/estina.c (hunk 1 above)
Comment (xplore93) [reply chall11/estina.c:52]: Shame on me

**Pull request merged**